cdap-common/src/main/java/co/cask/cdap/common/entity/EntityExistenceEnsurer.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.entity;
+
+import co.cask.cdap.common.NotFoundException;
+import co.cask.cdap.proto.id.EntityId;
+
+/**
+ * Interface to check if an {@link EntityId} exists in CDAP.
+ */
+public interface EntityExistenceEnsurer {
```

Comment:
I like the first one - `EntityExistenceVerifier`.
Without `Existence` (the 3rd and 4th option), it makes it seem like you're verifying/validating attributes of the entity, instead of just its existence.